rust_bindings/src/lib.rs

```diff
@@ -0,0 +1,132 @@
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+#![allow(dead_code)]
+#![allow(non_upper_case_globals)]
+
+use std::ffi::{CStr, CString};
+
+extern crate link_cplusplus;
+
+mod bindings {
+    include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+}
+
+pub fn create_discriminant(seed: &[u8], disc_size_bits: i32) -> Option<CString> {
+    // SAFETY: The length is guaranteed to match the actual length of the char pointer.
+    unsafe {
+        let ptr = bindings::create_discriminant_wrapper(seed.as_ptr(), seed.len(), disc_size_bits);
+        if ptr.is_null() {
+            return None;
+        }
+        let c_str = CStr::from_ptr(ptr).to_owned();
+        bindings::free(ptr as *mut std::ffi::c_void);
+        Some(c_str)
+    }
+}
+
+pub fn verify_n_wesolowski(
+    discriminant: &CStr,
+    x_s: &[u8],
+    proof: &[u8],
+    num_iterations: u64,
+    disc_size_bits: u64,
+    recursion: u64,
+) -> bool {
+    // SAFETY: The lengths are guaranteed to match the actual lengths of the char pointers.
+    unsafe {
+        let value = bindings::verify_n_wesolowski_wrapper(
+            discriminant.as_ptr() as *const std::ffi::c_char,
```

Comment:
so, if I understand correctly, this must be a null-terminated string, right?

Reply:
yes, but you cast it to a `c_char*`, which isn't necessarily a C string

Reply:
I got rid of this anyways